src/robot/libraries/Remote.py

```diff
@@ -260,6 +263,21 @@ def make_connection(self, host):
         self._connection = host, httplib.HTTPConnection(chost, timeout=self.timeout)
         return self._connection[1]
 
+class TimeoutHTTPSTransport(xmlrpclib.Transport):
+
+    def __init__(self, use_datetime=0, timeout=None):
+        xmlrpclib.Transport.__init__(self, use_datetime)
+        if not timeout:
+            timeout = socket._GLOBAL_DEFAULT_TIMEOUT
+        self.timeout = timeout
+
+    def make_connection(self, host):
+        if self._connection and host == self._connection[0]:
+            return self._connection[1]
+        chost, self._extra_headers, x509 = self.get_host_info(host)
+        self._connection = host, httplib.HTTPSConnection(chost, timeout=self.timeout)
```

Comment:
Do I get it right that the only difference between `TimeoutHTTPTransport` and `TimeoutHTTPSTransport` is that on this line the former uses `httplib.HTTPConnection` while the latter uses `httplib.HTTPSConnection` ? In that case there's a lot of unnecessary duplication.

One simple way to avoid duplication would be adding a class variable like `_connection_class =  httplib.HTTPConnection` to `TimeoutHTTPTransport` and then changing this line in that class to:

```python
self._connection = host, self._connection_class(chost, timeout=self.timeout)
```

That would allow changing `TimeoutHTTPSTransport` implementation to this:

```python
class TimeoutHTTPSTransport(TimeoutHTTPTransport):
    _connection_class = httplib.HTTPSConnection
```